### PR TITLE
[BEAM-5698] Migrate Go integration test to use Dataflow worker jar

### PR DESCRIPTION
**Pull request by herohde**
Also delete dead xml file.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by herohde** [comment]
run go postcommit

**Comment by herohde** [comment]
run go postcommit

**Comment by herohde** [comment]
run go postcommit

**Comment by herohde** [comment]
run go postcommit

**Comment by herohde** [comment]
R: @boyuanzz 

Website failure is unrelated.

File: sdks/go/test/run_integration_tests.sh [hunk 1]
```diff
@@ -79,7 +82,8 @@ echo ">>> RUNNING DATAFLOW INTEGRATION TESTS"
     --environment_config=$CONTAINER:$TAG \
     --staging_location=$GCS_LOCATION/staging-validatesrunner-test \
     --temp_location=$GCS_LOCATION/temp-validatesrunner-test \
-    --worker_binary=./sdks/go/test/build/bin/linux-amd64/worker
+    --worker_binary=./sdks/go/test/build/bin/linux-amd64/worker \
+    --dataflow_worker_jar=$DATAFLOW_WORKER_JAR
```
Comment (HuangLED) [inline sdks/go/test/run_integration_tests.sh]: I assume this integration test wants to trigger portable runner code path (apparently GO does not even have legacy world).

If that is the case, why triggering a job via portable runner does not require a docker image in the command line? versus the HOW-TO https://g3doc.corp.google.com/cloud/dataflow/portability/g3doc/howto.md?cl=head

I had the hypothesis that, maybe under the hood, what is inside JAR is essentially equivalent to what is in docker image. But if that is the case, why don't we stick with using docker image.  

What am I missing?

Comment (HuangLED) [reply sdks/go/test/run_integration_tests.sh]: Resolved.  

Looks like the docker image is in environment_config.

**Pull request merged**